mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -150,6 +150,26 @@ sealed trait Vector extends Serializable {
       toDense
     }
   }
+
+  /**
+   * Find the index of a maximal element.  Returns the first maximal element in case of a tie.
+   * Returns -1 if vector has length 0.
+   */
+  private[spark] def findMax: Int = {
```

Comment:
Oh, good point.  I'll move it to DenseVector.  We've often added helper methods in random places, rather than to linear algebra, utils, or the place they really should belong.  I prefer we add these methods directly to the relevant place but make them private.